File: app/src/main/java/com/o4x/musical/imageloader/glide/targets/AbsImageBitmapTarget.java
package com.o4x.musical.imageloader.glide.targets;

import android.graphics.Bitmap;
import android.widget.ImageView;

import androidx.annotation.Nullable;

import com.bumptech.glide.request.target.ImageViewTarget;
import com.o4x.musical.imageloader.glide.targets.palette.AbsPaletteTargetListener;

public class AbsImageBitmapTarget extends ImageViewTarget<Bitmap> {

    protected final AbsPaletteTargetListener listener;

    public AbsImageBitmapTarget(ImageView view, AbsPaletteTargetListener paletteTargetListener) {
        super(view);
        this.listener = paletteTargetListener;
    }

    @Override
    protected void setResource(@Nullable Bitmap resource) {
        view.setImageBitmap(resource);
        listener.onResourceReady(resource);
    }
}

File: app/src/main/java/com/o4x/musical/imageloader/glide/targets/AbsImageBitmapTarget.kt
package com.o4x.musical.imageloader.glide.targets

import android.graphics.Bitmap
import android.graphics.drawable.Drawable
import android.widget.ImageView
import androidx.core.graphics.drawable.toBitmap
import com.bumptech.glide.request.target.ImageViewTarget
import com.o4x.musical.imageloader.glide.targets.palette.AbsPaletteTargetListener

class AbsImageBitmapTarget(view: ImageView?, private val listener: AbsPaletteTargetListener) :
    ImageViewTarget<Bitmap?>(view) {

    override fun setDrawable(drawable: Drawable?) {
        super.setDrawable(drawable)
        listener.onResourceReady(drawable?.toBitmap(1, 1))
    }

    override fun setResource(resource: Bitmap?) {
        view.setImageBitmap(resource)
        listener.onResourceReady(resource)
    }
}